File: ValidParentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> string = new Stack<Character>();
        char[] vals = s.toCharArray();
        if(vals.length % 2 != 0){
            return false;
        }
        for(int i = 0; i < vals.length; i++){
            if(vals[i] == '('){
                string.push(')');
            }
            else if(vals[i] == '{'){
                string.push('}');
            }
            else if(vals[i] == '['){
                string.push(']');
            }
            else if(string.isEmpty() || string.pop() != vals[i]){
                return false;
            }
        }
        if(string.isEmpty()){
            return true;
        }
        return false;
    }
}

File: ValidParentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        for i in s:
            if i == '(' or i == '{' or i == '[':
                stack.append(i)
            else:
                if(len(stack) == 0):
                    return False
                elif i == ')':
                    if(stack.pop() != '('):
                        return False
                elif i == '}':
                    if(stack.pop() != '{'):
                        return False
                else:
                    if(stack.pop() != '['):
                        return False
        return len(stack) == 0
        